src/main/kotlin/telegram/Menu.kt

```diff
@@ -195,65 +169,211 @@ internal fun showRevealMenu(game: Game, bot: Bot, chatId: Long, messageId: Long)
+            row {
+                button(menuCommand named "Назад", messageId, 0, false)
+                if (itemsOffset != 0) {
+                    button(menuCommand named "Найти", messageId, itemsOffset - 1, false)
+                }
+            }
+        }
+    } else {
+        inlineKeyboard {
+            button(blankCommand named title)
+            if (listSize == 0) {
+                button(blankCommand named "🤷 Здесь ничего нет")
+            } else {
+                val pageIndex = itemsOffset / pageSize
+                val totalAvailablePages = listSize / pageSize +
+                        if (listSize % pageSize == 0) 0
+                        else 1
+                button(blankCommand named "Номер страницы: ${pageIndex + 1}")
+                val topItemIndex = topItemIndex(itemsOffset, pageSize)
+                row {
+                    if (pageIndex > 0) {
+                        button(menuCommand named "⬅", messageId, topItemIndex - pageSize, showNumpadMenu)
+                    }
+                    if (pageIndex < totalAvailablePages - 1) {
+                        button(menuCommand named "➡", messageId, topItemIndex + pageSize, showNumpadMenu)
+                    }
+                }
+                subList.forEachIndexed { index, item ->
+                    actionForEach(topItemIndex + index, item)
+                }
+                if (totalAvailablePages > 1) {
+                    row {
+                        button(menuCommand named "⏪ Первая", messageId, 0, showNumpadMenu)
+                        button(menuCommand named "⏩ Последняя", messageId, (totalAvailablePages - 1) * pageSize, showNumpadMenu)
+                    }
+                }
+                button(menuCommand named "Найти элемент по номеру", messageId, 0, true)
+            }
+            button(bottomButtonCommand, messageId)
+        }
+    }
     bot.editMessageReplyMarkup(
         ChatId.fromId(chatId),
         messageId,
-        replyMarkup = inlineKeyboard {
-            button(blankCommand named "Ведущие")
-            hostSettings.find().forEach {
-                button(chooseHostAdminCommand named (it.host?.fullName()?: ""), messageId, it.hostId)
-            }
-            button(adminBackCommand, messageId)
-        }
+        replyMarkup = markup
     )
 }
 
-internal fun showChosenSettingsMenu(chatId: Long, messageId: Long, bot: Bot, chosenId: Long) {
+internal fun showHostAdminSettingsMenu(
+    bot: Bot,
+    chatId: Long,
+    messageId: Long,
+    itemsOffset: Int,
+    showNumpadMenu: Boolean
+) {
+    val hostSettingsList = hostSettings.find()
+    showPaginatedMenu(
+        chatId,
+        messageId,
+        bot,
+        "Ведущие",
+        subListFromOffset(hostSettingsList, itemsOffset, defaultPageSize),
+        hostSettingsList.size,
+        { index, hostSettings ->
+            button(
+                chooseHostAdminCommand named "${index + 1}. ${hostSettings.host?.fullName()?: ""}",
```

Comment:
Number for no reason again

Reply:
Why are you quoting other comment here? This one is about numbers that take screen space with no actual use